src/main/java/io/confluent/idesidecar/restapi/credentials/ScramCredentials.java

```diff
@@ -0,0 +1,121 @@
+package io.confluent.idesidecar.restapi.credentials;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import io.confluent.idesidecar.restapi.exceptions.Failure;
+import io.confluent.idesidecar.restapi.exceptions.Failure.Error;
+import io.soabase.recordbuilder.core.RecordBuilder;
+import jakarta.validation.constraints.NotNull;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+@Schema(description = "Scram authentication credentials")
+@RecordBuilder
+public record ScramCredentials(
+    @Schema(description = "Hash algorithm")
+    @JsonProperty(value = "hash_algorithm")
+    @NotNull
+    HashAlgorithm hashAlgorithm,
+
+    @JsonProperty(value="scram_username")
+    @NotNull
+    String username,
+    @JsonProperty(value="scram_password")
+    @NotNull
+    Password password
```

Comment:
What about adding `@Schema` annotations to these two attributes, too? This way, we could provide a description and specify the expected `minLength` and `maxLength` in the OpenAPI spec ([see how ApiKeyAndSecret uses it](https://github.com/confluentinc/ide-sidecar/blob/main/src/main/java/io/confluent/idesidecar/restapi/credentials/ApiKeyAndSecret.java#L27-L28)); note that `@Schema` impacts only the generated OpenAPI spec not the behavior of the application.

Reply:
thanks! I'm going to set similar lengths to the api key and secret as well, as I couldn't find anything in[ RFC 5802](https://datatracker.ietf.org/doc/html/rfc5802) on the length of them